tools/syz-headerparser/headerparser.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2015 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+'''
+This scripts takes as input a list of header files, a device name and generates metadata
+files to make syzkaller device descriptions.
+'''
+
+import argparse
+import logging
+
+from headerlib.container import GlobalHierarchy
+
+
+def main():
+    """
+    python parser.py --device=test --filename=A.h,B.h
+    """
+
+    parser = argparse.ArgumentParser(description='Parse header files to output fuzzer'
+                                                 'struct metadata.')
+    parser.add_argument('--device',
+                        help='name of the device you want to write a fuzzer for',
+                        dest='device',
+                        required=True)
+    parser.add_argument('--filenames',
+                        help='comma-separated header filenames',
+                        dest='filenames',
+                        required=True)
+    parser.add_argument('--debug',
+                        help='print debug-information at every level of parsing',
+                        action='store_true')
+    parser.add_argument('--template',
+                        help='add lines from <template> into the template in headerpreprocessor',
```

Comment:
I, as a user, don't understand this description.
What is a template? Does it have to be written in some template language? Which?
What do I pass as the value? text? or file?
What is headerpreprocessor?
What template in headerpreprocessor?

Let's call this `-include`, because that the standard flag name in gcc and clang for this function. Also it better describes the purpose of this flag:
```
    parser.add_argument('--include',
                        help='include the specified file as the first line of the processed header files',
```